### ISSUE TITLE
Problem with R 4.1.0 and catboost 0.26 - "Binary classification loss/metrics require label data" 

### ISSUE THREAD

**Issue by anilorak-ibk** [catboost/catboost#1739]
Problem: Binary classification loss/metrics require label data
catboost version: 0.26
Operating System: Windows Server 2016 64-bit

Hi there! 

I recently had to update my R and RStudio versions. I am currently using R 4.1.0 and RStudio 1.4.1106 
I am trying to train a catboost model with a code that used to work until few months ago. Today, I am getting this error and cannot find a solution for it: 

model fit failed for Fold3: depth=4, learning_rate=0.1077, iterations=500, l2_leaf_reg=12, rsm=1, border_count=149 Error in catboost.train(pool, test_pool, param) : 
  C:/Program Files (x86)/Go Agent/pipelines/BuildMaster/catboost.git/catboost/private/libs/target/data_providers.cpp:585: **Binary classification loss/metrics require label data**
- Fold3: depth=4, learning_rate=0.1077, iterations=500, l2_leaf_reg=12, rsm=1, border_count=149 
There were missing values in resampled performance measures.Aggregating results
Something is wrong; all the ROC metric values are missing:
      ROC           Sens          Spec    
 Min.   : NA   Min.   : NA   Min.   : NA  
 1st Qu.: NA   1st Qu.: NA   1st Qu.: NA  
 Median : NA   Median : NA   Median : NA  
 Mean   :NaN   Mean   :NaN   Mean   :NaN  
 3rd Qu.: NA   3rd Qu.: NA   3rd Qu.: NA  
 Max.   : NA   Max.   : NA   Max.   : NA  
 NA's   :1     NA's   :1     NA's   :1    
Error: Stopping

It is indeed a binary classification, but output data (label) is available in the model. Also, data do not contain any missing values. This is the training code I am using (for if it helps...). 

model = caret::train(y = train$label, 
                  x = train[,-1], # first column is label
                  method = my.catboost.caret,
                  preProc = NULL,
                  tuneGrid = train_params, 
                  trControl = train_control,
                  class_weights = c(1, labels_summary["output_0"] / labels_summary["output_1"]),
                  has_time =TRUE,
                  random_seed = random_seed,
                  use_best_model = TRUE,
                  logging_level = "Silent", 
                  od_type = "Iter",
                  od_wait = 30,
                  eval_metric = "AUC",
                  test_pool = eval_pool
  )
  
 I cannot really post a replicable code here, because it is a complicated code that I didnt write myself - and that uses some self-written functions. But as I said, the code was working just few months ago. 
 
 Can somebody help me? Has somebody encountered a similar error - and solved it? 
 
 Thank you very much for your help!!

**Comment by anilorak-ibk** [comment]
**UPDATE: I figured out the problem was caused by the catboost interface** (see https://github.com/catboost/catboost/blob/master/catboost/R-package/R/catboost.caret.R). 
The interface calls the function catboost.from_data_frame(), which is deprecated. I could make the code work by using an old package version of catboost (which I guess contains the catboost.from_data_frame() function). 

What confuses me a bit is that the interface code was last updated on February 2021 (or at least, the last commit was done by then) but as I said, it cannot be used with new package versions of catboost! 

I hope this can help someone out there :) 

Cheers!
